### ISSUE TITLE
Ignore regular expression character set

### ISSUE THREAD

**Issue by cuixiping** [streetsidesoftware/vscode-spell-checker#691]
A character set in regular expression should not be considered as a word.  
For example,  `/[asdw]/` actually means `/(a|s|d|w)/`, should be ignored.

**Comment by Jason3S** [comment]
I agree with you. The challenge is detecting regular expressions.
On work around is to add a comment to ignore the word or line:

```
// cspell:ignore asdw
// or
// cspell:ignore-next-line
r = /[asdw]/gi
```

**Comment by cuixiping** [comment]
```
// cspell:disable-next-line
hello = /world_[asdw]/gi;  // some comments
```

Other parts of the line need spell checking. Only the [ ] pattern need be ignored.

**Comment by cuixiping** [comment]
I tried to ignore all regular expression but failed. The following setting has no effect.

```
{
    "cSpell.ignoreRegExpList": [
        "\\\\/.+?\\\\/"
    ]
}
```

I don't want the "MLHV" to be marked as unknown word.

```
var regex= /[MLHV]/gi;
```

**Comment by Jason3S** [comment]
I suggest using the VS Code Settings UI to edit the expression, it will be easier to understand.

![image](https://user-images.githubusercontent.com/3740137/131619102-41e63e5c-55a8-4014-b0e4-2edf8306c706.png)

The top expression `/\/.+?\/[gimu]*/g` will work. The bottom expression I got from your example, it will not match a RegExp.

I suggest using a website like: [Example on regex101.com](https://regex101.com/r/Z3yKFr/1) to try out your expressions.

Note: the `g` is important or the spell checker will only match the first occurrence.

**Comment by cuixiping** [comment]
It works. Thank you very much.

```
{
    "cSpell.ignoreRegExpList": [
        "/\\/.+?\\/[gimusyd]*/gm"
    ]
}
```
